handlers/ExecuteViewSubmitHandler.ts

```diff
@@ -26,20 +34,211 @@ export class ExecuteViewSubmitHandler {
+                        let branch = share_branch_record[0]['share_github_branch_input'] as string
+                        let accessToken = await getAccessTokenForUser(this.read, user, this.app.oauth2Config);
+                        
+                        if (!accessToken) {
+                            await sendNotification(this.read, this.modify, user, room, "Please make sure your OAuth2 settings are ready and logged into Github using  \`/ai-programmer login \` !");
+                        } else {
+                            let getresponse = await getNewCode(this.http, repo, path, input_str, commit, accessToken?.token, branch);
+                            
+                            if (getresponse && !getresponse?.serverError) {
+                                
+                                await sendNotification(this.read,this.modify,user,room,`File exists, current content will be overwritten!`);
+                                
+                                let response = await uploadNewCode(
+                                    this.http, repo, path, input_str, commit, accessToken?.token, branch, getresponse.sha
+                                )
+                                if(response && !response?.serverError){
+                                    await sendNotification(this.read,this.modify,user,room,`Successfully shared your code to Github!`);
+                                }else{
+                                    await sendNotification(this.read,this.modify,user,room,`Sharing to Github failed!`);
+                                }
+                            }
+                            else {
+                                
+                                let response = await uploadNewCode(
+                                    this.http, repo, path, input_str, commit, accessToken?.token, branch
+                                )
+                                if(response && !response?.serverError){
+                                    await sendNotification(this.read,this.modify,user,room,`Successfully shared your code to Github!`);
+                                }else{
+                                    await sendNotification(this.read,this.modify,user,room,`Sharing to Github failed!`);
+                                }
+                            }
+                            
+                        }
+                    } else {
+                        await sendNotification(this.read, this.modify, user, room, "You have to enter all information!");
+                    }
+                }
+                catch(err){
+                    
+                }
                 break;
             }
-            case Modals.GEN_ACTION: {
-                console.log("gen_action view submit");
+            case 'shareGist': {
+                try {
+                    const persis = this.read.getPersistenceReader();
```

Comment:
check whether user is logged into GitHub first.

Reply:
added function